robot/src/main/kotlin/org/team9432/Robot.kt

```diff
@@ -0,0 +1,158 @@
+            1 -> Logger.recordMetadata("GitDirty", "Uncomitted changes")
+            else -> Logger.recordMetadata("GitDirty", "Unknown")
+        }
+
+        if (isReal() || mode == Mode.SIM) {
+            Logger.addDataReceiver(WPILOGWriter()) // Log to a USB stick ("/U/logs")
+            Logger.addDataReceiver(NT4Publisher()) // Publish data to NetworkTables
+            PowerDistribution(1, PowerDistribution.ModuleType.kRev) // Enables power distribution logging
+        } else if (mode == Mode.REPLAY) {
+            setUseTiming(false) // Run as fast as possible
+            val logPath = LogFileUtil.findReplayLog() // Pull the replay log from AdvantageScope (or prompt the user)
+            Logger.setReplaySource(WPILOGReader(logPath)) // Read replay log
+            Logger.addDataReceiver(
+                WPILOGWriter(
+                    LogFileUtil.addPathSuffix(
+                        logPath,
+                        "_sim"
+                    )
+                )
+            ) // Save outputs to a new log
+        }
+
+        // Logger.disableDeterministicTimestamps() // See "Deterministic Timestamps" in the "Understanding Data Flow" page
+        Logger.start()
+
+        Logger.recordOutput("Subsystems/Climber", Pose3d(Translation3d(0.0, 0.0, 0.0), Rotation3d()))
+        Logger.recordOutput(
+            "Subsystems/Limelight",
+            Pose3d(Translation3d(-0.063500, 0.0, 0.420370 + 0.124460), Rotation3d(0.0, 0.0, Math.toRadians(180.0)))
+        )
+
+        PortForwarder.add(5800, "photonvision.local", 5800)
+
+        HAL.report(FRCNetComm.tResourceType.kResourceType_Language, FRCNetComm.tInstances.kLanguage_Kotlin)
+
+        Controls
+        Vision
+        Gyro
+        Drivetrain
+        Hopper
+        Intake
+        Hood
+        Shooter
+        Amp
+        Beambreaks
+        LeftClimber
+        RightClimber
+        AdditionalTriggers
+        AutoBuilder
+        AutoChooser
+    }
+
+    override fun robotPeriodic() {
+        KCommandScheduler.run()
+        RobotState.log()
+
+        LEDState.updateState()
+
+        Logger.recordOutput("Drive/FuturePose", RobotPosition.getFuturePose(CommandConstants.SHOOT_ON_MOVE_SECS))
+        Logger.recordOutput("SpeakerSide", RobotPosition.getSpeakerSide())
+
+        DriverStation.getAlliance().getOrNull()?.let { alliance = it }
+    }
+
+    override fun disabledInit() {
+        KCommandScheduler.cancelAll()
+    }
+
+    override fun autonomousInit() {
+        ParallelCommand(
+            AutoChooser.getCommand(),
+            HoodAimAtSpeaker()
+        ).schedule()
+    }
+
+    override fun teleopInit() {
+        stop()
+
+        if (RobotState.noteInSpeakerSideHopperBeambreak()) {
+            SequentialCommand(
+                PullFromSpeakerShooter(),
+                InstantCommand { RobotState.hasRemainingAutoNote = true }
+            ).withTimeout(0.75).schedule()
+        }
```

Comment:
I can't recall if there was a time we still shot the preload after not shooting it in auto. I could go back and watch matches to find out if this did happen. Are we confident in our ability to do this? Or do we need to add this as a ToDo in Trello?

Reply:
this is more for if we a-stop it with a note in the hopper

Reply:
Makes sense, does the first shot in tele still work if we end up a-stopping with the note in the hopper? Only match I can recall off the top of my head where this may have happened was our last qual on blue where the auto drove into the wall. But there isn't a video of it.